pravega/pravega
# Issue 2188: (SegmentStore) Split the common Thread Pool into Core and Storage

**Change log description**
Split the Common SegmentStore Thread Pool in two:
- Storage Pool: every operation executed in `Storage` (essentially what `SyncStorage` is run on). This is exclusively for the IO intensive operations that the Storage adapters run (all Storage adapters are non-async, and the `AsyncStorageWrapper` uses a thread pool to run those tasks on). 
- Core Pool: everything else (except BookKepeer and Netty - those haven't changed).
    - This includes `StorageWriter`, which is a SegmentStore component that picks data from `DurableLog` and issues Storage requests (while the StorageWriter executes on the Core pool, the Storage requests are run on the Storage Pool (see bullet above)).

Updated some callbacks to better ensure that they are executed on the proper thread pool (in `StreamSegmentMapper` and `SegmentAggregator`).

**Purpose of the change**
Fixes #2188.

**What the code does**
See Description. There are no functional changes to the SegmentStore. The bulk of the logic is in `ServiceBuilder`.

**How to verify it**
All unit tests must pass. Examine the run of a System test and verify that all Tier2 operations are executed on their special thread pool and that there aren't too many non-Tier2 Storage messages on that pool (exceptional messages will end up there since exception handlers are executed in the same thread).
